OpenMarket/OpenMarket/Controller/ProductDetailViewController.swift

```diff
@@ -0,0 +1,134 @@
+//
+//  ProductDetailViewController.swift
+//  OpenMarket
+//
+//  Created by 웡빙, 보리사랑 on 2022/08/03.
+//
+
+import Foundation
+import UIKit
+
+final class ProductDetailViewController: UIViewController {
+    var productDetailView: ProductDetailView?
+    var productId: Int?
+    var viewControllerTitle: String?
+    // MARK: - View Life Cycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.view.backgroundColor = .systemBackground
+        setupNavigationItem()
+        productDetailView = ProductDetailView(self)
+        productDetailView?.horizontalScrollView.delegate = self
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        guard let productId = productId else {
+            return
+        }
+        NetworkManager.shared.requestProductDetail(at: productId) { detail in
+            DispatchQueue.main.async { [weak self] in
+                self?.updateSetup(with: detail)
+            }
+        }
+    }
+    // MARK: - VC Method
+    private func setupNavigationItem() {
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .action, target: self, action: #selector(setupBarButtonTap))
+        navigationItem.title = self.viewControllerTitle
+    }
+    
+    private func updateSetup(with detail: ProductDetail) {
+        guard let productDetailView = productDetailView else { return }
+        detail.images.forEach { image in
+            let imageView = ProductImageView(width: Int(self.view.frame.width) - 20, height: Int(self.view.frame.width*0.7))
+            imageView.setImageUrl(image.url)
+            imageView.contentMode = .scaleAspectFit
+            productDetailView.horizontalStackView.addArrangedSubview(imageView)
+        }
+        productDetailView.productNameLabel.text = detail.name
+        productDetailView.stockLabel.text = "남은수량 : \(detail.stock)"
```

Comment:
```suggestion
        productDetailView.stockLabel.text = detail.stockLabelText
```
이렇게 따로 메소드나 연산프로퍼티로 만들어 뷰에 값 할당하면 재사용성이 늘어나 좋습니다. 
이러면 Presenter의 역할을 하므로 detail은 뷰모델의 역할을 하게된 셈입니다(따지자면 데이터바인딩까지 해야 뷰모델입니다).

Reply:
오 이렇게 사용하는 방법도 생각해보지 못했는데 정말 좋은것 같습니다. 
그리고 궁금한점이, 이렇게 사용되는 text값들을 각각의 모델이 연산프로퍼티로 소유하는것이 아니라 Namespace로 묶어서 사용하는 방법은 별로 좋지 않은 방법일까요?

[8ce1738](https://github.com/yagom-academy/ios-open-market/pull/201/commits/8ce1738aae56004d9f646e692f74e1755de1815a)

Reply:
네임스페이스로 묶어서 사용하는 방법이 무엇인지 잘 몰라서 그러는데 혹시 예시를 보여주실 수 있을까요?!

ps. 그리고 모델이 연산 프로퍼티를 가지는건 프로퍼티처럼 소유하는건 아니고 매번 접근될때마다 계산해서 반환하므로 저는 메소드의 성격이 더 있다고 생각합니다 :)